File: 2021_Algs_train_1.0/homework_2/task_C.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func Abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func solve(arr []int, x int) int {
	minDiffs := make([]int, len(arr))
	for i := 0; i < len(arr); i++ {
		minDiffs[i] = Abs(arr[i] - x)
	}
	
	minDiff := minDiffs[0]
	index := 0
	for i := 0; i < len(arr); i++ {
		if minDiffs[i] < minDiff {
			minDiff = minDiffs[i];
			index = i
		}
	}
	return arr[index]
}

func main() {
	// Create a scanner to read from standard input
	scanner := bufio.NewScanner(os.Stdin)

	// Read the first line containing a single integer
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())

	// Read the second line containing space-separated integers
	scanner.Scan()
	arrStr := strings.Split(scanner.Text(), " ")
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		arr[i], _ = strconv.Atoi(arrStr[i])
	}

	// Read the third line containing a single integer
	scanner.Scan()
	x, _ := strconv.Atoi(scanner.Text())

	fmt.Println(solve(arr, x))
}

File: 2021_Algs_train_1.0/homework_2/task_C.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int n,x;
    scanf("%d",&n);
    int *arr = malloc(sizeof(int)*n);
    for (int i=0; i<n; i++) {
        scanf("%d",&arr[i]);
    }
    scanf("%d",&x);

    int *min_diffs = malloc(sizeof(int)*n);
    for (int i=0; i<n; i++) {
        min_diffs[i] = abs(arr[i] - x);
    }

    int min_diff = min_diffs[0];
    int index = 0;
    for (int i=0; i<n; i++) {
        if (min_diffs[i] < min_diff) {
            min_diff = min_diffs[i];
            index = i;
        }
    }
    printf("%d", arr[index]);
    free(min_diffs);
    free(arr);
    return 0;
}